007-extension-flag.md

```diff
@@ -0,0 +1,57 @@
+| Title  | Implement --extension flag  |
+|--------|-----------------------------|
+| Author | @bmeck                      |
+| Status | DRAFT                       |
+| Date   | 2017-08-09T10:00:00-05:00   |
+
+## Description
+
+The Node CLI currently accepts input in both filename (`node app.js`) and opaque text stream form (`node -e '123'` and `node <app.js`). This behavior always results in running the stream as a CommonJS goal. With the upcoming usage of ESM and WASM there will be cases where developers wish to pipe other forms of text to the CLI.
```

Comment:
`Script goal` and maybe also `other program forms to the CLI`

Reply:
On second thought, I suppose that last one could also be clarified as `source text` if that is more consistent.